### PR TITLE
Update "Behavior changes"

### PR DESCRIPTION
**Pull request by jtcohen6**
## What are you changing in this pull request and why?

- Rename "Legacy behaviors" to "Behavior changes" — this is how we've been referring to them
- Add definition of what we consider to be a behavior change, as well as examples of what does / doesn't constitute a behavior change

### CODE REVIEW

File: website/docs/reference/global-configs/behavior-changes.md [hunk 1]
```diff
@@ -12,10 +12,31 @@ Another category of flags provides existing projects with a migration window for
...
+1. **Introduction (disabled by default):** dbt adds logic to support both 'old' and 'new' behaviors. The 'new' behavior is gated behind a flag, disabled by default, preserving the old behavior.
 2. **Maturity (enabled by default):** The default value of the flag is switched, from `false` to `true`, enabling the new behavior by default. Users can preserve the 'old' behavior and opt out of the 'new' behavior by setting the flag to `false` in their projects. They may see deprecation warnings when they do so.
 3. **Removal (generally enabled):** After marking the flag for deprecation, we remove it along with the 'old' behavior it supported from the dbt codebases. We aim to support most flags indefinitely, but we're not committed to supporting them forever. If we choose to remove a flag, we'll offer significant advance notice.
 
+## What is a behavior change?
+
+> The same dbt project code and same dbt commands return one result before the behavior change, and they would return a different result after the behavior change.
+
+Examples of behavior changes:
```
Comment (jtcohen6) [inline website/docs/reference/global-configs/behavior-changes.md:23]: @MichelleArk Do these examples look reasonable & correct to you?

**Review by dbeatty10 — APPROVED** [review]

File: website/docs/docs/dbt-versions/core-upgrade/07-upgrading-to-v1.8.md [hunk 2]
```diff
@@ -98,13 +98,13 @@ The ability for installed packages to override built-in materializations without
...
 
 You can read more about each of these behavior changes in the following links:
 
-- (Mature, enabled by default) [Require explicit package overrides for builtin materializations](/reference/global-configs/legacy-behaviors#require_explicit_package_overrides_for_builtin_materializations)
-- (Introduced, disabled by default) [Require resource names without spaces](https://docs.getdbt.com/reference/global-configs/legacy-behaviors#require_resource_names_without_spaces)
-- (Introduced, disabled by default) [Run project hooks (`on-run-*`) in the `dbt source freshness` command](/reference/global-configs/legacy-behaviors#source_freshness_run_project_hooks)
+- (Mature, enabled by default) [Require explicit package overrides for builtin materializations](/reference/global-configs/behavior-changes#require_explicit_package_overrides_for_builtin_materializations)
+- (Introduced, disabled by default) [Require resource names without spaces](https://docs.getdbt.com/reference/global-configs/behavior-changes#require_resource_names_without_spaces)
```
Comment (nataliefiann) [inline website/docs/docs/dbt-versions/core-upgrade/07-upgrading-to-v1.8.md]:
```suggestion
- (Introduced, disabled by default) [Require resource names without spaces](/reference/global-configs/behavior-changes#require_resource_names_without_spaces)
```

File: website/docs/reference/global-configs/behavior-changes.md (hunk 1 above)
Comment (MichelleArk) [reply website/docs/reference/global-configs/behavior-changes.md:23]: Yes these are excellent!

**Review by MichelleArk — APPROVED** [review]

File: website/docs/reference/global-configs/behavior-changes.md [hunk 3]
```diff
@@ -12,10 +12,31 @@ Another category of flags provides existing projects with a migration window for
...
+- dbt begins raising a validation _error_ that it didn't previously.
+- dbt changes the signature of a built-in macro. Your project has a custom reimplementation of that macro. This could lead to errors, because your custom reimplementation will be passed arguments it cannot accept.
+- A dbt adapter renames or removes a method that was previously available on the `{{ adapter }}` object in the dbt-Jinja context.
+- dbt makes a breaking change to contracted metadata artifacts by deleting a required field, changing the name or type of an existing field, or removing the default value of an existing field ([README](https://github.com/dbt-labs/dbt-core/blob/37d382c8e768d1e72acd767e0afdcb1f0dc5e9c5/core/dbt/artifacts/README.md#breaking-changes))
```
Comment (nataliefiann) [inline website/docs/reference/global-configs/behavior-changes.md]:
```suggestion
- dbt makes a breaking change to contracted metadata artifacts by deleting a required field, changing the name or type of an existing field, or removing the default value of an existing field ([README](https://github.com/dbt-labs/dbt-core/blob/37d382c8e768d1e72acd767e0afdcb1f0dc5e9c5/core/dbt/artifacts/README.md#breaking-changes)).
```

File: website/docs/reference/global-configs/behavior-changes.md [hunk 4]
```diff
@@ -12,10 +12,31 @@ Another category of flags provides existing projects with a migration window for
...
+## What is a behavior change?
+
+> The same dbt project code and same dbt commands return one result before the behavior change, and they would return a different result after the behavior change.
+
+Examples of behavior changes:
+- dbt begins raising a validation _error_ that it didn't previously.
+- dbt changes the signature of a built-in macro. Your project has a custom reimplementation of that macro. This could lead to errors, because your custom reimplementation will be passed arguments it cannot accept.
+- A dbt adapter renames or removes a method that was previously available on the `{{ adapter }}` object in the dbt-Jinja context.
+- dbt makes a breaking change to contracted metadata artifacts by deleting a required field, changing the name or type of an existing field, or removing the default value of an existing field ([README](https://github.com/dbt-labs/dbt-core/blob/37d382c8e768d1e72acd767e0afdcb1f0dc5e9c5/core/dbt/artifacts/README.md#breaking-changes)).
+- dbt removes one of the fields from [structured logs](/reference/events-logging#structured-logging)
```
Comment (nataliefiann) [inline website/docs/reference/global-configs/behavior-changes.md]:
```suggestion
- dbt removes one of the fields from [structured logs](/reference/events-logging#structured-logging).
```

File: website/docs/reference/global-configs/behavior-changes.md [hunk 5]
```diff
@@ -12,10 +12,31 @@ Another category of flags provides existing projects with a migration window for
...
+- A dbt adapter renames or removes a method that was previously available on the `{{ adapter }}` object in the dbt-Jinja context.
+- dbt makes a breaking change to contracted metadata artifacts by deleting a required field, changing the name or type of an existing field, or removing the default value of an existing field ([README](https://github.com/dbt-labs/dbt-core/blob/37d382c8e768d1e72acd767e0afdcb1f0dc5e9c5/core/dbt/artifacts/README.md#breaking-changes)).
+- dbt removes one of the fields from [structured logs](/reference/events-logging#structured-logging).
+
+The following are **not** behavior changes:
+- Fixing a bug where the previous behavior was defective, undesirable, or undocumented.
+- dbt begins raising a _warning_ that it didn't previously.
+- dbt updates the language of human-friendly messages in log events.
+- dbt makes a non-breaking change to contracted metadata artifacts by adding a new field with a default, or deleting a field with a default ([README](https://github.com/dbt-labs/dbt-core/blob/37d382c8e768d1e72acd767e0afdcb1f0dc5e9c5/core/dbt/artifacts/README.md#non-breaking-changes))
```
Comment (nataliefiann) [inline website/docs/reference/global-configs/behavior-changes.md]:
```suggestion
- dbt makes a non-breaking change to contracted metadata artifacts by adding a new field with a default, or deleting a field with a default ([README](https://github.com/dbt-labs/dbt-core/blob/37d382c8e768d1e72acd767e0afdcb1f0dc5e9c5/core/dbt/artifacts/README.md#non-breaking-changes)).
```

**Review by nataliefiann — COMMENTED** [review]
Hiya @jtcohen6 

Thanks for creating this PR - it's great! Other than a few punctuation amendments and one broken link (which I updated) I have no comments or suggestions. I haven't approved this yet as it looks like you're awaiting an additional review from some PMs. Please let me know if there's anything else I can do to help with this.

Kind Regards

File: website/docs/reference/global-configs/behavior-changes.md [hunk 6]
```diff
@@ -12,10 +12,31 @@ Another category of flags provides existing projects with a migration window for
...
+1. **Introduction (disabled by default):** dbt adds logic to support both 'old' and 'new' behaviors. The 'new' behavior is gated behind a flag, disabled by default, preserving the old behavior.
 2. **Maturity (enabled by default):** The default value of the flag is switched, from `false` to `true`, enabling the new behavior by default. Users can preserve the 'old' behavior and opt out of the 'new' behavior by setting the flag to `false` in their projects. They may see deprecation warnings when they do so.
 3. **Removal (generally enabled):** After marking the flag for deprecation, we remove it along with the 'old' behavior it supported from the dbt codebases. We aim to support most flags indefinitely, but we're not committed to supporting them forever. If we choose to remove a flag, we'll offer significant advance notice.
 
+## What is a behavior change?
+
+> The same dbt project code and same dbt commands return one result before the behavior change, and they would return a different result after the behavior change.
```
Comment (matthewshaver) [inline website/docs/reference/global-configs/behavior-changes.md]:
```suggestion
The same dbt project code and the same dbt commands return one result before the behavior change, and they return a different result after the behavior change.
```

**Review by matthewshaver — APPROVED** [review]

**Comment by jtcohen6** [comment]
Thanks everyone for the review & help getting this over the line! @MichelleArk @dbeatty10 @nataliefiann @matthewshaver

**Pull request merged**